keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -437,8 +434,8 @@ def train_step(self, data):
         gradients = tape.gradient(loss, trainable_vars)
         self.optimizer.apply_gradients(zip(gradients, trainable_vars))
 
-        # Early exit for no train time metrics
-        if not self.evaluate_train_time_metrics:
+        # Early exit for no custom metrics
```

Comment:
we should still gate this behind a constructor or compile() argument, in the GPU case you only want the COCO metrics on a tiny subset of your eval metrics